routes/user.go

```diff
@@ -1811,12 +1811,87 @@ type GetNotificationsRequest struct {
 	FilteredOutNotificationCategories map[string]bool
 }
 
+type GetNotificationsCountRequest struct {
+	PublicKeyBase58Check string
+}
+
+
 type GetNotificationsResponse struct {
 	Notifications       []*TransactionMetadataResponse
 	ProfilesByPublicKey map[string]*ProfileEntryResponse
 	PostsByHash         map[string]*PostEntryResponse
 }
 
+type GetNotificationsCountResponse struct {
+	NotificationsCount uint64
+}
+
+func (fes *APIServer) GetNotificationsCount(ww http.ResponseWriter, req *http.Request) {
+	decoder := json.NewDecoder(io.LimitReader(req.Body, MaxRequestBodySizeBytes))
+	requestData := GetNotificationsCountRequest{}
+	if err := decoder.Decode(&requestData); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"GetNotificationsCount: Problem parsing request body: %v", err))
+		return
+	}
+
+	userMetadata, err := fes.getUserMetadataFromGlobalState(requestData.PublicKeyBase58Check)
+
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetNotificationsCount: Error getting user metadata from global state: %v", err))
+		return
+	}
+
+	var fetchStartIndex int64
+
+	if userMetadata.LatestUnreadNotificationIndex > 0 {
+		fetchStartIndex = userMetadata.LatestUnreadNotificationIndex + 1
+	} else if userMetadata.NotificationLastSeenIndex > 0 {
+		fetchStartIndex = userMetadata.NotificationLastSeenIndex + 1
+	}
```

Comment:
Oh, it's interesting that you are using both of these.  Why do we need both?

Reply:
If we've already run `getUnreadNotificationsCount` before, we want to start where we left off last time so as to not duplicate work (the result from the last scan will be stored in `UnreadNotifications`). If `getUnreadNotificationsCount` has not been called since the user last checked their notifications, we want to continue from the last notification the user has read.